sirepo/package_data/template/raydata/parameters.py.jinja

```diff
@@ -1,11 +1,32 @@
 # -*- python -*-
-from pykern import pksubprocess
+import os
+import papermill
+from pykern import pkio
+from pykern.pkdebug import pkdlog, pkdexc
 
-pksubprocess.check_call_with_signals((
-    'papermill',
-    # POSIT: same kernel isntalled by rscode-ipykernel
-    '--kernel=py3',
-    '--log-output',
-    '{{ input_name }}',
-    '{{ output_name }}',
-))
+def _main():
+    try:
+        # Creating the directory acts as a lock. If we are able to
+        # create the dir then we get the "lock". If the dir already
+        # exists then another agent has it locked or the analysis is
+        # completed.
+        os.makedirs('{{ scan_dir }}')
+    except FileExistsError:
+        return
```

Comment:
Ideally we'd want to never get here (lock in postgres) and for the user to then get the state in the db. W/o postgres we could use state files and we could read it here but that adds new overhead I'm not sure is worth it. With the code as it is, the state will always be reported back completed. That is a bug because the state may not be completed (running or error). I'm not sure what the best implementation is. Thoughts?

Reply:
If we are working under the model that the system for the phase 1 really only works with one user then we will never get here (we will get the sate from the supervisor) and this is good enough. We really don't even need any locking in that case...

Reply:
c926e4820